x-pack/legacy/plugins/lens/public/indexpattern_plugin/indexpattern_suggestions.ts

```diff
@@ -354,9 +365,49 @@ function createNewLayerWithMetricAggregation(
 export function getDatasourceSuggestionsFromCurrentState(
   state: IndexPatternPrivateState
 ): Array<DatasourceSuggestion<IndexPatternPrivateState>> {
+  const layers = Object.entries(state.layers || {});
+  if (layers.length > 1) {
+    // Return suggestions that reduce the data to each layer individually
+    return layers
+      .map(([layerId, layer], index) => {
+        const hasMatchingLayer = layers.some(
+          ([otherLayerId, otherLayer]) =>
+            !(otherLayerId === layerId) && otherLayer.indexPatternId === layer.indexPatternId
```

Comment:
Nit: `!(otherLayerId === layerId)` is much clearer when written `otherLayerId !== layerId`